akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -277,15 +269,6 @@ abstract class AbstractActor extends Actor {
   @throws(classOf[Exception])
   override def postStop(): Unit = super.postStop()
 
-  // TODO In 2.6.0 we can remove deprecation and make the method final
-  @deprecated("Override preRestart with message parameter with Optional type instead", "2.5.0")
-  @throws(classOf[Exception])
-  @silent
-  override def preRestart(reason: Throwable, message: Option[Any]): Unit = {
-    import scala.compat.java8.OptionConverters._
-    preRestart(reason, message.asJava)
```

Comment:
we must still call this, so the change should be to make this method `final` and remove the `@deprecated`

Reply:
@patriknw I meant to ask about he comment. Why must we still call this and not super.preRestart?

Reply:
because user should be able to override
```
def preRestart(reason: Throwable, message: Optional[Any]): Unit
```

note the `Optional` parameter difference

and with this change we never call that method

Reply:
Aha, gotcha

Reply:
This gets tricky when mixed with stashing, i'll revert this for now

Reply:
I see